src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java

```diff
@@ -292,6 +291,27 @@ public void mouseClicked(MouseEvent e) {
                     popup.setSize(450, 450);
                     popup.setLocation(e.getX(), e.getY());
                     popup.setVisible(true);
+                    popup.addKeyListener(new KeyAdapter() {
```

Comment:
Yeah this sounds good let me change this!

Reply:
However, a quick question for the `play/pause` button should we keep it this way or we also change it to like the action for play/pause button? Because the action for the `play/pause` button is not as simple as the `stepBack` and `stepForward`

Reply:
Yeah, I think we could reuse the `playPauseButtonListener` for the key press event too.

Reply:
Okay that sounds good.

Reply:
Generally we want something like:
```
UI Component 1 ---
                   \___ Business Logic
                   /
UI Component 2 ---/
```
Rather than:
```
UI Component 1 --- UI Component 2 --- Business Logic
```

Reply:
Okay so this way it is easier to make changes on only one part of the UI in the future instead of whole branch?

Reply:
Yes, it means we could change the '+1', '-1' and other buttons work without breaking the keyboard shortcut feature. For example, we might want to replace the buttons with a jog shuttle. It's not an immediate problem, just a cleaner design not to couple one part of the GUI to another part of the GUI.

Can we also pull out the key event listener into a single instance and reuse that for the popup windows and the main window?

Reply:
Yes I will try to change that.